src/main/java/frc/robot/CommandSwerveDrivetrain.java

```diff
@@ -55,6 +64,55 @@ private void startSimThread() {
     m_simNotifier.startPeriodic(kSimLoopPeriod);
   }
 
+  // xSpeed, ySpeed, rotationSpeed should be axes with range -1<0<1
+  public SwerveRequest getDriveRequest(double xSpeed, double ySpeed, double rotationSpeed) {
+    xSpeed *= maxSpeed;
+    ySpeed *= maxSpeed;
+    rotationSpeed *= maxAngularRate;
+    if (isFieldOriented) {
+      if (alignmentRotation != null) {
+        return new SwerveRequest.FieldCentricFacingAngle()
+            .withTargetDirection(alignmentRotation)
+            .withDriveRequestType(DriveRequestType.Velocity)
+            .withVelocityX(xSpeed)
+            .withVelocityY(ySpeed);
+      }
+      return new SwerveRequest.FieldCentric()
+          .withDriveRequestType(DriveRequestType.Velocity)
+          .withVelocityX(xSpeed)
+          .withVelocityY(ySpeed)
+          .withRotationalRate(rotationSpeed);
+    }
+    return new SwerveRequest.RobotCentric()
+        .withDriveRequestType(DriveRequestType.Velocity)
+        .withVelocityX(xSpeed)
+        .withVelocityY(ySpeed)
+        .withRotationalRate(rotationSpeed);
+  }
+
+  public SwerveRequest getBrakeRequest() {
+    return new SwerveRequest.SwerveDriveBrake();
+  }
+
+  public SwerveRequest getAlignRequest(Rotation2d rotation) {
+    return new SwerveRequest.FieldCentricFacingAngle().withTargetDirection(rotation);
+  }
+
+  public void toggleOrientation() {
+    isFieldOriented = !isFieldOriented;
+    if (!isFieldOriented) {
```

Comment:
do you need an 
```
else:
    setAlignment(wayToGetAlignment);
```

Reply:
Worth adding a comment to show intention (robot oriented drive means we don't want to/can't hold a field oriented heading)